src/staging/src/components/megaMenu.js

```diff
@@ -1,20 +1,267 @@
-import React from "react"
+import React, { useState, useEffect, useCallback } from "react";
+import { Link, navigate } from "gatsby"
 import PropTypes from "prop-types"
-import "../styles/megaMenu/desktop.scss"
-import "../styles/megaMenu/mobile.scss"
-import MobileMegaMenu from "./megaMenu/mobileMenu.js"
-import DesktopMegaMenu from "./megaMenu/desktopMenu.js"
+import { isTablet } from "react-device-detect"
+import "../styles/megaMenu/megaMenu.scss"
+
+const MegaMenu = ({ content, menuMode }) => {
+
+  const [menuContent, setMenuContent] = useState([]) 
+  const [menuTree, setMenuTree] = useState([])
+  const [menuRoot, setMenuRoot] = useState({})
+  const [selectedItem, setSelectedItem] = useState([]) // most recent item user has interacted with
+  const [selections, setSelections] = useState({}) // the selected item at each level, i.e. selection breadcrumbs
+  const [isMenuOpen, setIsMenuOpen] = useState(false) // currently only used for mobile - menu closed at first
+
+  const getSelectionObj = (item, obj) => {
+    // this creates an object that indicates all the pages that
+    // are "selected" based on the selectedItem and its parents
+    // i.e. the breadcrumbs back to home
+    // this allows us to highlight the selected parents and
+    // keep the correct menus open
+    if (item.strapiId === 1) {
+      // have reached home, 
+      // add home at level 0 and return, ending he recursion
+      obj[0] = item;
+      return (obj)
+    } else {
+      // add this item as the one that is "selected" at its level
+      // then recurse to its parent to add the next highest level
+      obj[item.treeLevel] = item;
+      obj = getSelectionObj(item.parent, obj) // recurse
+      return (obj)
+    }
+  }
+
+  const menuNavigate = (item) => {
+    // either open the menu for the item's children (isPageNav=false)
+    // or navigate to the associated url (isPageNav=true)
+    if (item) {
+      let isPageNav = false;
+
+      if (!item.hasChildren) {
+        isPageNav = true;
+      }
+      if (item === selectedItem) {
+        // reselect
+        isPageNav = true;
+      }
+
+      if (isPageNav) {
+        navigate(item.url)
+        menuReset()
+      } else {
+        setSelectedItem(item)
+        let selObj = getSelectionObj(item, {}); // track the selected item at this level and above
+        setSelections(selObj)
+      }
+    }
+  }
+
+  const navigateBack = (e, item) => {
+    // go "up" a level by navigating to this item's parent
+    menuNavigate(item.parent)
+  }
+
+  const navigatePage = (e, item, menuMode) => {
+    if (menuMode === 'sitemap') {
+      navigate(item.url)
+    } else {
+      menuNavigate(item, menuMode)
+    }
+  }
+
+  const sectionHover = (e, section, menuMode) => {
+    if ((window.screen.width >= 992) && (menuMode !== 'sitemap') && !isTablet) { // otherwise hover triggered in mobile emulator
```

Comment:
Something strange here - hover select still happens on screens <922 wide, if mobile emulator is not active.

Reply:
that's true, didn't consider that scenario, will disable hover for that case

Reply:
turns out i should have been using window.innerWidth, fixed